### PR TITLE
12475

### PR DESCRIPTION
**Pull request by MongoCaleb**
# IMPORTANT
Because this PR also includes code in another PR, please **IGNORE ALL FILES** except `access-custom-user-data.txt`!
### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-12475

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/realm/caleb/12475/dotnet/access-custom-user-data.html

### CODE REVIEW

File: examples/dotnet/Examples/CustomUserDataExamples.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        MongoClient.Collection<CustomUserData> cudCollection;
+
+
+        [OneTimeSetUp]
+        public async Task Creates()
+        {
+            // :code-block-start: create
+            app = App.Create(myRealmAppId);
+            user = await app.LogInAsync(Credentials.EmailPassword("foo@foo.com", "foobar"));
+            var config = new SyncConfiguration("ufda", user);
```
Comment (nirinchev) [inline examples/dotnet/Examples/CustomUserDataExamples.cs:30]: Should these be in the example code block?

File: source/dotnet/access-custom-user-data.txt [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+document containing the user ID of the currently logged in user and a
+``favoriteColor`` value into the custom user data collection:
+
+.. literalinclude:: /examples/generated/code/start/CustomUserDataExamples.codeblock.create.cs
+   :language: csharp
+
+You can add any number of arbitrary fields and values to the custom user
+data document when you create it. The user ID field is the only
+required fields. You may find it helpful to create a C# class (POCO) that 
+represents the custom user data. UYou can then serialize/deserialize this class
```
Comment (nirinchev) [inline source/dotnet/access-custom-user-data.txt:58]:
```suggestion
represents the custom user data. You can then serialize/deserialize this class 
```

File: examples/dotnet/Examples/CustomUserDataExamples.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        {
+            await cudCollection.DeleteManyAsync();
+        }
+
+    }
+
+    // :code-block-start: cud
+    public class CustomUserData
+    {
+        [Required]
```
Comment (nirinchev) [inline examples/dotnet/Examples/CustomUserDataExamples.cs:94]: `Required` has no effect on this class as it's a RealmObject attribute.

File: examples/dotnet/Examples/CustomUserDataExamples.cs [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        public string _id { get; set; }
```
Comment (nirinchev) [inline examples/dotnet/Examples/CustomUserDataExamples.cs:95]: Probably make the setter private since we only want to set it in the constructor?

**Review by nirinchev — COMMENTED** [review]
Looking good! On a non-docs related note, do you think I should change the `CustomData` property to be instead `GetCustomData` method with a generic overload:

```csharp
public BsonDocument GetCustomData() {}
public T GetCustomData<T>() {}
```

File: examples/dotnet/Examples/CustomUserDataExamples.cs (hunk 1 above)
Comment (nirinchev) [reply examples/dotnet/Examples/CustomUserDataExamples.cs:30]: But the sync config should at least be removed, right? And if we're keeping the login part, let's use non-foo email/password :P

File: source/dotnet/access-custom-user-data.txt (hunk 2 above)
Comment (MongoCaleb) [reply source/dotnet/access-custom-user-data.txt:58]: oops

**Comment by MongoCaleb** [comment]
> do you think I should change the `CustomData` property to be instead `GetCustomData` method with a generic overload:
> 
> ```cs
> public BsonDocument GetCustomData() {}
> public T GetCustomData<T>() {}
> ```

That would be sick! How it is now is also easy/great, but being able to deserialize in one call would be nice to have, for sure!

**Review by nirinchev — APPROVED** [review]
Looks good - only https://github.com/mongodb/docs-realm/pull/542#discussion_r504299181 remains to be addressed + will need a few code changes once I make `CustomData` into a method (sorry!)

**Review by ccho-mongodb — APPROVED** [review]
LGTM! Left some nitpicks and questions.

**Pull request merged**